src/NetAnalyzers/Core/Microsoft.NetCore.Analyzers/Security/Helpers/InsecureDeserializationTypeDecider.cs

```diff
@@ -334,9 +334,13 @@ ImmutableArray<InsecureObjectGraphResult> Compute((ITypeSymbol, ObjectGraphOptio
                     foreach (AttributeData knownTypeAttributeData in typeSymbol.GetAttributes(this.KnownTypeAttributeTypeSymbol))
                     {
                         if (knownTypeAttributeData.AttributeConstructor.Parameters.Length != 1
-                            || knownTypeAttributeData.ConstructorArguments.Length != 1
-                            || knownTypeAttributeData.ConstructorArguments[0] is not TypedConstant typedConstant
-                            || typedConstant.Kind != TypedConstantKind.Type    // Not handling the string methodName overload
+                            || knownTypeAttributeData.ConstructorArguments.Length != 1)
+                        {
+                            continue;
+                        }
+
+                        TypedConstant typedConstant = knownTypeAttributeData.ConstructorArguments[0];
+                        if (typedConstant.Kind != TypedConstantKind.Type    // Not handling the string methodName overload
```

Comment:
❔ Can we just go back to the original form here?

Reply:
Yes, sure I can do that as well. However, seemed like an unnecessary is expression check. @dotpaul What do you prefer?

Reply:
I have reverted the changes with a source suppression.